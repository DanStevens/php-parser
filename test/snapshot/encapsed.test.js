const parser = require('../main');

describe("encapsed", function() {
  it("variable (simple syntax)", function() {
    expect(parser.parseEval('echo "string $var string";')).toMatchSnapshot();
  });
  it("two variable (simple syntax)", function() {
    expect(parser.parseEval('echo "string $var->$var string";')).toMatchSnapshot();
  });
  it.only("variable curly (simple syntax)", function() {
    expect(parser.parseEval('echo "string ${var} string";')).toMatchSnapshot();
  });
  it("offsetlookup (simple syntax)", function() {
    expect(parser.parseEval('echo "string $array[0] string";')).toMatchSnapshot();
  });
  it("offsetlookup (simple syntax)", function() {
    expect(parser.parseEval('echo "string $array[koolaid1] string";')).toMatchSnapshot();
  });
  it("propertylookup (simple syntax)", function() {
    expect(parser.parseEval('echo "string $obj->property string";')).toMatchSnapshot();
  });
});
