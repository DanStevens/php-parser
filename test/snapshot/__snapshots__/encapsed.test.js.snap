// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encapsed offsetlookup (simple syntax) 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string $array[0] string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            OffsetLookup {
              "kind": "offsetlookup",
              "offset": Number {
                "kind": "number",
                "value": "0",
              },
              "what": Variable {
                "byref": false,
                "curly": false,
                "kind": "variable",
                "name": "array",
              },
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed offsetlookup (simple syntax) 2`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string $array[koolaid1] string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            OffsetLookup {
              "kind": "offsetlookup",
              "offset": Identifier {
                "kind": "identifier",
                "name": "koolaid1",
              },
              "what": Variable {
                "byref": false,
                "curly": false,
                "kind": "variable",
                "name": "array",
              },
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed propertylookup (simple syntax) 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string $obj->property string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            PropertyLookup {
              "kind": "propertylookup",
              "offset": Identifier {
                "kind": "identifier",
                "name": "property",
              },
              "what": Variable {
                "byref": false,
                "curly": false,
                "kind": "variable",
                "name": "obj",
              },
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed two variable (simple syntax) 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string $var->$var string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            Variable {
              "byref": false,
              "curly": false,
              "kind": "variable",
              "name": "var",
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "->",
              "unicode": false,
              "value": "->",
            },
            Variable {
              "byref": false,
              "curly": false,
              "kind": "variable",
              "name": "var",
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed variable (simple syntax) 1`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string \${var} string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            Variable {
              "byref": false,
              "curly": true,
              "kind": "variable",
              "name": Identifier {
                "kind": "identifier",
                "name": "var",
              },
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed variable (simple syntax) 2`] = `
Program {
  "children": Array [
    Echo {
      "expressions": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string \${var} string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            Variable {
              "byref": false,
              "curly": true,
              "kind": "variable",
              "name": Identifier {
                "kind": "identifier",
                "name": "var",
              },
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`encapsed variable curly (simple syntax) 1`] = `
Program {
  "children": Array [
    Echo {
      "arguments": Array [
        Encapsed {
          "kind": "encapsed",
          "raw": "\\"string \${var} string\\"",
          "type": "string",
          "value": Array [
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": "string ",
              "unicode": false,
              "value": "string ",
            },
            Variable {
              "byref": false,
              "curly": true,
              "kind": "variable",
              "name": "var",
            },
            String {
              "isDoubleQuote": false,
              "kind": "string",
              "raw": " string",
              "unicode": false,
              "value": " string",
            },
          ],
        },
      ],
      "kind": "echo",
      "shortForm": false,
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
